File: opendaylight/rest/src/main/resources/org/opendaylight/controller/rest/Activator.scala
package org.opendaylight.controller.rest

import org.osgi.framework.BundleActivator
import org.osgi.framework.BundleContext

class Activator extends BundleActivator {
    def start(context: BundleContext) {
      println("REST Service started")
    }
    
    def stop(context: BundleContext) {}
}

File: opendaylight/northbound/clustering/src/main/java/org/opendaylight/controller/clustering/Activator.java
package org.opendaylight.controller.clustering;

import org.opendaylight.controller.clustering.northbound.ClusteringNorthbound;
import org.osgi.framework.BundleActivator;
import org.osgi.framework.BundleContext;
import org.osgi.framework.ServiceRegistration;

public class Activator implements BundleActivator {
   private ServiceRegistration registration;

   @Override
   public void start(BundleContext context) throws Exception {
      ClusteringNorthbound cnb = new ClusteringNorthbound();
      registration = context.registerService(
      ClusteringNorthbound.class.getName(), cnb, null);
   }

   @Override
   public void stop(BundleContext context) throws Exception {
      registration.unregister();
   }
}
